pkg/clustermanager/reconcile_clusterprovider.go

```diff
@@ -99,14 +99,14 @@ func (c *controller) handleAdd(provider *lcv1alpha1.ClusterProviderDesc) error {
 }
 
 func (c *controller) handleDelete(providerName string) error {
-	err := c.StopDiscovery(providerName)
+	ns := c.providers[providerName].nameSpace
```

Comment:
This will panic if the key is not in the map.

Reply:
fixed